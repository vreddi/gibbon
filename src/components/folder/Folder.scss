.gibbon-folder {
  display: flex;
  flex-direction: column;
  justify-content: center;
  width: 8em;

  .gibbon-folder-image {
    cursor: pointer;
  }

  .gibbon-folder-label-text {
    color: white;
    cursor: text;
  }

  .gibbon-folder-label-input {
    font-size: 1em;
    text-align: center;
    color: white;
    outline: none;
    background-color: transparent;
    border: none;
    width: 100%;
  }
}
