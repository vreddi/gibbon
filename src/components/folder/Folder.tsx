import React, { Component } from 'react';
import Props from './Folder.props';
import State from './Folder.state';

import './Folder.scss';

class Folder extends Component<Props, State> {

  constructor(props: Props) {
    super(props);

    this.state = {
      label: props.label || "Gibbon",
      isEditMode: false
    };
  }

  getFolder = (): JSX.Element => {
    return (
      <svg className="gibbon-folder-image" version="1" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 48 48" enableBackground="new 0 0 48 48">
        <path fill="#63a4ff" d="M40,12H22l-4-4H8c-2.2,0-4,1.8-4,4v8h40v-4C44,13.8,42.2,12,40,12z"/>
        <path fill="#1976d2" d="M40,12H8c-2.2,0-4,1.8-4,4v20c0,2.2,1.8,4,4,4h32c2.2,0,4-1.8,4-4V16C44,13.8,42.2,12,40,12z"/>
      </svg>
    );
  }

  getLabelField = (): JSX.Element => {
    if (this.state.isEditMode) {
      return this.getLabelInputField();
    }

    return this.getLabelTextField();
  }

  getLabelTextField = (): JSX.Element => {
    return (
      <span
        data-testid="folder-label"
        className="gibbon-folder-label-text"
        onClick={() => this.setState({ isEditMode: true })}
      >
        {this.state.label}
      </span>
    );
  }

  getLabelInputField = () => {
    return (
      <input
        data-testid="folder-input"
        autoFocus
        className="gibbon-folder-label-input"
        value={this.state.label}
        onChange={e => this.setState({ label: e.target.value })}
        onKeyPress={(event) => {
          event.preventDefault();

          if (event.key === 'Enter') {
            this.setState({isEditMode: false});
          }
        }}
      />
    );
  }

  render(): JSX.Element {
    return (
      <div data-testid="folder" className="gibbon-folder">
        {this.getFolder()}
        {this.getLabelField()}
      </div>
    );
  }
}

export default Folder;
