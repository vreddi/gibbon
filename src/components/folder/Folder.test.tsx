import React from 'react';
import ReactDOM from 'react-dom';
import Folder from './Folder';
import { render, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';

import "@testing-library/jest-dom/extend-expect";

describe('📁 Folder', () => {

  afterEach(() => {
    cleanup();
  })

  test('can render', () => {
    ReactDOM.render(<Folder />, document.createElement('div'));
  });

  test('contains correct elements', () => {
    const { getByTestId } = render(<Folder label="Football highlights" />)

    expect(getByTestId("folder")).toBeDefined();
    expect(getByTestId("folder-label")).toBeDefined();

    expect(getByTestId("folder-label")).toHaveTextContent("Football highlights");
  });

  test('matches snapshot', () => {
    const tree = renderer.create(<Folder label="Football highlights" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
})