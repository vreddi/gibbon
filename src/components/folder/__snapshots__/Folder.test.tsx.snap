// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`📁 Folder matches snapshot 1`] = `
<div
  className="gibbon-folder"
  data-testid="folder"
>
  <svg
    className="gibbon-folder-image"
    enableBackground="new 0 0 48 48"
    version="1"
    viewBox="0 0 48 48"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M40,12H22l-4-4H8c-2.2,0-4,1.8-4,4v8h40v-4C44,13.8,42.2,12,40,12z"
      fill="#63a4ff"
    />
    <path
      d="M40,12H8c-2.2,0-4,1.8-4,4v20c0,2.2,1.8,4,4,4h32c2.2,0,4-1.8,4-4V16C44,13.8,42.2,12,40,12z"
      fill="#1976d2"
    />
  </svg>
  <span
    className="gibbon-folder-label-text"
    data-testid="folder-label"
    onClick={[Function]}
  >
    Football highlights
  </span>
</div>
`;
